### PR TITLE
Bug 1242905 - React component for job list platform names

### PR DESCRIPTION
**Pull request by caseywilliams**
Two changes:

1. Installs [`ng-react-test-utils`](https://github.com/jrwebdev/ngreact-test-utils) to provide a reusable compile() function for ngreact tests. Previously I just wrote a one-off `compile()` in the revision list test file, but reusing it presented a problem. Setting up browserify for the react tests only seemed like the path of least resistance - if this seems like overkill, I'll see if I can figure something else out.
2. Adds a simple platform name component to start the migration of the job list table to react.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2055)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by camd** [comment]
I noticed that when the no job list takes the full screen width, then it doesn't get the correct left-side justification.  See this: https://www.dropbox.com/s/m0w1l7r3zya3vp4/Screenshot%202017-01-05%2016.41.07.png?dl=0

Locally, you can test this by just setting the "Filter by Platform and jobs" field to "build".

File: ui/js/directives/treeherder/clonejobs.js [hunk 1]
```diff
@@ -974,22 +971,20 @@ treeherder.directive('thCloneJobs', [
...
-                    }
-                );
-                rowHtml += '<td class="job-row">' + getJobTableRowHTML(platform.groups) + '</td></tr>';
-                tableHtml += rowHtml;
+                var scope = _.extend($rootScope.$new(), {
+                    'name': thPlatformName(platform.name),
+                    'option': platform.option,
+                    'id': platformId
+                });
+                var platformTdEl = $compile('<jobplatformtd watch-depth="reference" id="id" name="name" option="option" watch-depth="reference"></jobplatformtd>')(scope);
```
Comment (camd) [inline ui/js/directives/treeherder/clonejobs.js]: ``watch-depth="reference"`` is in there twice.  :)

File: ui/js/directives/treeherder/clonejobs.js [hunk 2]
```diff
@@ -958,9 +955,9 @@ treeherder.directive('thCloneJobs', [
         };
         var generateJobElements = function(resultsetAggregateId, resultset) {
```
Comment (camd) [inline ui/js/directives/treeherder/clonejobs.js]: Lots of function in this file were poorly named.  Probably by me at some point...  :)  This function might be more aptly named as ``generateJobTable``.  Or can you think of a better name?

**Comment by caseywilliams** [comment]
Thanks @camd! I think everything's been fixed now, but let me know if not, of course.

**Pull request merged**